gobblin-completeness/src/main/java/org/apache/gobblin/completeness/verifier/KafkaAuditCountVerifier.java

```diff
@@ -125,10 +126,13 @@ public Map<CompletenessType, Boolean> calculateCompleteness(String datasetName,
     countsByTier.forEach((x,y) -> log.info(String.format(" %s : %s ", x, y)));
 
     Map<CompletenessType, Boolean> result = new HashMap<>();
-    result.put(CompletenessType.ClassicCompleteness, calculateCompleteness(datasetName, beginInMillis, endInMillis,
-        CompletenessType.ClassicCompleteness, countsByTier) > threshold);
-    result.put(CompletenessType.TotalCountCompleteness, calculateCompleteness(datasetName, beginInMillis, endInMillis,
-        CompletenessType.TotalCountCompleteness, countsByTier) > threshold);
+    Arrays.stream(CompletenessType.values()).forEach(type -> {
+      try {
+        result.put(type, calculateCompleteness(datasetName, beginInMillis, endInMillis, type, countsByTier) > threshold);
+      } catch (IOException e) {
+        log.error("Failed to calculate completeness for type " + type, e);
```

Comment:
Should we set a value to false if its computation fails? So that the returned map always contains two elements.
Just to avoid null value because `ClassicWatermarkUpdater` & `TotalCountWatermarkUpdater` have similar logic like:

> protected void computeAndUpdateInternal(Map<KafkaAuditCountVerifier.CompletenessType, Boolean> results,
        ZonedDateTime timestampDT) {
      if (!results.get(KafkaAuditCountVerifier.CompletenessType.TotalCountCompleteness)) {
        return;
      }

Reply:
Yes good point, I missed that it would cause a null failure there. Instead of always returning false, I changed the checks themselves to handle null properly, that seems safer in case the map is incomplete for some reason.
`if (!results.getOrDefault(KafkaAuditCountVerifier.CompletenessType.ClassicCompleteness, false))`